common/OpTestSystem.py

```diff
@@ -141,6 +145,13 @@ def set_state(self, state):
         self.state = state
 
     def goto_state(self, state):
+        # only perform detection when incoming state is UNKNOWN
+        # if user overrides from command line and machine not at desired state can lead to exceptions
+        if (self.state == OpSystemState.UNKNOWN):
+          print "OpTestSystem CHECKING CURRENT STATE and TRANSITIONING for TARGET STATE: %s" % (state)
+          self.state = self.run_DETECT(state)
+          print "OpTestSystem CURRENT DETECTED STATE: %s" % (self.state)
+
```

Comment:
And also consider fixing flash testcases. when user doing flash test, don't run run_DETECT, continue the test by doing power off and flash fw and power on, this already does by testcases/OpTestFlash test module. So only in this case you need a way to skip run_DETECT.

Reply:
> testcases/OpTestFlash

I reviewed OpTestFlash and with the latest patch this should perform optimally.

To clarify a little, when a test suite is started there is only a single time where the incoming self.state will equal UNKNOWN.  If this case happens, unless the target state desired is a flip of either Petitboot to OS OR OS to Petitboot the logic falls through and nothing is done.  So I think the cases are ok (but maybe I'm missing something).

You can review the output to see what happens, in the snippet below run_DETECT is invoked as a function call, but then the logic falls through and nothing is done other than starting the test run as truly UNKNOWN=0.  

OpTestSystem CHECKING CURRENT STATE and TRANSITIONING for TARGET STATE: 2
OpTestSystem CURRENT DETECTED STATE: 0

So maybe try the patch and then send me the flows or a recreate to illustrate the concern to help me clarify.

Reply:
i think now we are okay with this patch for OpTestFlash scenarios, as it always starts the test with goto_state(OFF).